themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,136 @@
+
+- Auto-Scaling: With serverless, resources automatically scale up or down to match the workload so you don't use infrastructure beyond what you actually need.
+
+- Pay-as-You-Go: You are charged only for the actual compute resources you use during code execution.
+
+The serverless architecture model is in line with current tech trends such as cloud-native development and DevOps, by allowing developers to focus on code development and pushing faster releases.
+
+## Serverless Function Design
+
+The design of the serverless function is important for the success of your application. When mapping out the architectural design for your Azure function, you should consider some of the following:
+
+- **Function Logic**: A serverless function should have a well-defined purpose and focus on a specific task. If a function has multiple responsibilities, consider breaking it into smaller, more focused functions. In the serverless app we deployed using Pulumi, we had two functions `date` and `weather` and it is clear what their responsibilities are.
+
+    ``` go
+    // The 'date' function that returns the current date
+	mux.HandleFunc("/api/date", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(fmt.Sprintf(`{ "now": %d }`, time.Now().UnixNano()/1000000)))
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		weather := getWeatherData()
+
+		// Convert the weather data to JSON.
+		responseJSON, err := json.Marshal(weather)
+		if err != nil {
+			http.Error(w, "Error encoding response to JSON", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+	})
+    ```
+
+- **Statelessness**: Serverless functions should be stateless. This encourages scalability and fault tolerance since every serverless function call should be able to function on its own, regardless of previous or subsequent calls. The recommended practise for managing data that has to be persistent beyond the span of one function execution is to store it externally in reliable data storage systems like object stores or databases. In doing so, serverless applications maintain the clear separation between function logic and stateful data management while guaranteeing data durability, consistency, and adaptability to changing workloads.
```

Comment:
Given that HTTP is (generally speaking) a stateless protocol, perhaps here is where the mention of serverless as a good fit for HTTP-based use cases would fit?